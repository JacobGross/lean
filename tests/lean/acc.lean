namespace play
inductive acc (A : Type) (R : A → A → Prop) :  A → Prop :=
intro : ∀ (x : A), (∀ (y : A), R y x → acc A R y) → acc A R x

check @acc.rec
end play
