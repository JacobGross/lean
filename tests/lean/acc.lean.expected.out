acc.rec :
  Π {A : Type} {R : A → A → Prop} {C : A → Type},
    (Π (x : A), (∀ (y : A), R y x → acc A R y) → (Π (y : A), R y x → C y) → C x) →
    (Π {a : A}, acc A R a → C a)
